### PR TITLE
New parameters in New-AzureRmVMSqlServerAutoBackupConfig cmdlet to support Auto Backup for SQL Server 2016 VMs.

### PR DESCRIPTION
**Pull request by anilkyelam**
## Description

1. Renamed New-AzureVMSqlServer* cmdlets to include "Rm" in the name. Kept the old ones for backward compatibility.
2. Added new parameters for New-AzureRmVMSqlServerAutoBackupConfig that reflect changes for Auto Backup V2 for SQL IaaS Extension.
3. Fixed bug in Get-AzureRmVMSqlServerExtension to correctly get values of parameters by parsing the substatus in the response. 

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThrough` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@anilyela__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (anilyela). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/NewAzureRmVMSqlServerAutoBackupConfig.cs [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    /// <summary>
+    /// Helper cmdlet to construct instance of AutoBackup settings class
+    /// </summary>
+    [Cmdlet(
+        VerbsCommon.New,
+        ProfileNouns.VirtualMachineSqlServerAutoBackupConfig,
+        DefaultParameterSetName = StorageUriParamSetName),
+    OutputType(
+        typeof(AutoBackupSettings))]
+    public class NewAzureRmVMSqlServerAutoBackupConfigCommand : AzureRMCmdlet
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/NewAzureRmVMSqlServerAutoBackupConfig.cs]: @anilyela if you are changing the name of the cmdlet `New-AzureVMSqlServerAutoBackupConfig` to `New-AzureRmVMSqlServerAutoBackupConfig`, you do not need to add an entirely new file - you can add an alias in the old cmdlet to `New-AzureRmVMSqlServerAutoBackupConfig`. There does not need to be separate files for the same cmdlet.

File: src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/NewAzureRmVMSqlServerAutoPatchingConfig.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+{
+    /// <summary>
+    /// Helper cmdlet to construct instance of AutoPatching settings class
+    /// </summary>
+    [Cmdlet(
+         VerbsCommon.New,
+         ProfileNouns.VirtualMachineSqlServerAutoPatchingConfig),
+     OutputType(
+         typeof(AutoPatchingSettings))]
+    public class NewAzureRmVMSqlServerAutoPatchingConfigCommand : PSCmdlet
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/NewAzureRmVMSqlServerAutoPatchingConfig.cs]: @anilyela same comment about adding alias

File: src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/NewAzureRmVMSqlServerKeyVaultCredentialConfig.cs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    /// <summary>
+    /// Helper cmdlet to construct instance of SQL Credential AKV based settings class
+    /// </summary>
+    [Cmdlet(
+         VerbsCommon.New,
+         ProfileNouns.VirtualMachineSqlServerKeyVaultCredentialConfig,
+         SupportsShouldProcess = true),
+     OutputType(
+         typeof(KeyVaultCredentialSettings))]
+    public class NewAzureRmVMSqlServerKeyVaultCredentialConfigCommand : PSCmdlet
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/NewAzureRmVMSqlServerKeyVaultCredentialConfig.cs]: @anilyela same comment about adding alias

File: src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/NewAzureVMSqlServerAutoBackupConfig.cs [hunk 4]
```diff
@@ -117,6 +120,77 @@ public SecureString StorageKey
             set;
         }
 
+        [Parameter(
+          Mandatory = false,
+          Position = 6,
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/NewAzureVMSqlServerAutoBackupConfig.cs]: @anilyela we ask that there be no more than four positional parameters for a cmdlet. Please remove the position attribute from the parameters you have added to this cmdlet.

File: src/ResourceManager/Compute/ChangeLog.md [hunk 5]
```diff
@@ -18,6 +18,8 @@
         - Additional information about change #1
 -->
 ## Current Release
+* New parameters in New-AzureRmVMSqlServerAutoBackupConfig cmdlet to support Auto Backup for SQL Server 2016 VMs.
+* New-AzureVMSqlServer* cmdlets are renamed to New-AzureRmVMSqlServer* now. Old ones will continue to work however.
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/ChangeLog.md]: @anilyela for the change log, can you talk about the new parameters in `New-AzureRmVMSqlServerAutoBackupConfig`?

```
* New parameters in ...
    - BackupSystemDbs: < brief description>
    - BackupScheduleType: <brief description>
```

File: src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/NewAzureVMSqlServerAutoBackupConfig.cs (hunk 4 above)
Comment (anilyela) [inline src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/NewAzureVMSqlServerAutoBackupConfig.cs]: done

**Comment by anilyela** [comment]
@cormacpayne addressed your comments.

**Comment by cormacpayne** [comment]
@anilyela are there any more sign-offs you are waiting on?

**Comment by anilyela** [comment]
@cormacpayne no, signoff from nimeshnedungadi is enough from our side.

**Comment by anilyela** [comment]
Thanks @cormacpayne. When is next release that takes change this live?

**Comment by cormacpayne** [comment]
@anilyela you can view the dates of the next three Azure PowerShell releases [here](https://github.com/Azure/azure-powershell/milestones)

It is scheduled for January 18th

**Pull request merged**